packages/cozy-harvest-lib/src/components/TriggerManager.jsx

```diff
@@ -51,17 +56,35 @@ export class TriggerManager extends Component {
    * @return {Object}          io.cozy.jobs document, runned with account data
    */
   async handleAccountCreationSuccess(account) {
-    const { createTrigger, konnector } = this.props
+    const {
+      createDirectoryByPath,
+      createTrigger,
+      fetchDirectoryByPath,
+      konnector,
+      t
+    } = this.props
 
     this.setState({
       createdAccount: account
     })
 
+    let folder
+    if (konnectors.needsFolder(konnector)) {
+      const path = `${t('default.baseDir')}/${konnector.name}/${slugify(
+        accounts.getLabel(account)
+      )}`
+
+      folder =
+        (await fetchDirectoryByPath(path)) ||
+        (await createDirectoryByPath(path))
+    }
```

Comment:
This can be extracted into a `ensureDirectoryForKonnector` method ? I think the need for `let` is a smell.

Reply:
I like the idea, but it's a little work, I will refactor it the next time I will implement some logic for files.